dockerfiles/evmapp/entrypoint.sh

```diff
@@ -42,21 +62,33 @@ else
     export HOME=/root
 fi
 
-# detect external IPv4 address
-SCNODE_NET_DECLAREDADDRESS="$(dig -4 +short +time=2 @resolver1.opendns.com A myip.opendns.com | grep -v ";" || true)"
+# Checking if external IP address is provided by the user via ENV var
+if [ -n "${SCNODE_NET_DECLAREDADDRESS:-}" ]; then
+  # Checking IPv(4|6) address validity
+  ipv4_pattern="^((25[0-5]|2[0-4][0-9]|1[0-9][0-9]|[1-9]?[0-9])\.){3}(25[0-5]|2[0-4][0-9]|1[0-9][0-9]|[1-9]?[0-9])$"
+  ipv6_pattern="^((([0-9a-fA-F]){1,4})\:){7}([0-9a-fA-F]){1,4}$"
```

Comment:
The regex doesn't support compressed ipv6 notation which both dig and curl return, e.g. `dead:beef::`. I'd suggest to utilize the `ipv6calc` utility (in apt) to validate v4 and v6 addresses as that is more robust.

Reply:
This check is only a defensive mechanism from USER input IF someone wants to provide IP address manually, but I can use this ipv6calc to make the check bullet proof :+1: